src/dynamics/video_player/player/player.js

```diff
@@ -585,10 +586,16 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
                             if (thisPlaylist.length === 1 && thisPlaylist[0].title === this.get('title')) return;
                             const showNextTime = Number(this.get("shownext")) || 0;
                             const engageTime = showNextTime + (Number(this.get("noengagenext")) || 0);
-                            const duration = this.get("duration")
-                            const shallActivateNext = (duration - showNextTime) >= (Number(this.get("noengagenext")) || 0)
-                            if (position > showNextTime && showNextTime && !this.get("next_active") && shallActivateNext) {
-                                this.set("next_active", true);
+                            const duration = this.get("duration");
+                            const remainingTime = (duration - position);
+                            const minIntervalNextWidget = this.get("nextwidgetminintervalbeforeend");
+                            const shallActivateNext = minIntervalNextWidget < 0 ? true :
```

Comment:
Hey @Josewd should we set `nextwidgetminintervalbeforeend` as some negative value? Or it's state required to be updated in somewhere in progress?